File: ThreadsApp/ThreadsApp/View/Chat/Cells/CellTypeHelper.m
//
//  CellTypeHelper.m
//  ThreadsApp
//
//  Created by Tabriz Dzhavadov on 13/07/2017.
//  Copyright © 2017 Brooma Service. All rights reserved.
//

#import "CellTypeHelper.h"
#import <Threads/Threads.h>
#import "PlistUtils.h"

@implementation CellTypeHelper

+ (NSString *) textByType: (CellType) type {
    if (type == CellTypeVersion) {
        return [NSString stringWithFormat: @"ThreadsLib - v%@\nThreadsApp - v%@", [Threads version], [PlistUtils getAppVersion]];
    } else if (type == CellTypeDebugMode) {
        return @"Debug-mode";
    } else if (type == CellTypeDesign) {
        return @"Design";
    } else if (type == CellTypeToFragmentChat) {
        return [NSString stringWithFormat: @"%@ %@",
                NSLocalizedString(@"show_chat_title", nil),
                NSLocalizedString(@"chat_fragment_title", nil)];
    } else if (type == CellTypeToFullChat) {
        return [NSString stringWithFormat: @"%@ %@",
                NSLocalizedString(@"show_chat_title", nil),
                NSLocalizedString(@"chat_full_controller_title", nil)];
    } else if (type == CellTypeToOtherChats) {
        return @"Other Chat Integrations";
    } else if (type == CellTypeToOutsideTextMessage) {
        return @"Outside Text Message";
    } else if (type == CellTypeToOutsideImageMessage) {
        return @"Outside Image Message";
    } else {
        return @"unknown type";
    }
}

@end

File: ThreadsApp/ThreadsAppSwift/View/Chat/Cells/CellTypeHelper.swift
//
//  CellTypeHelper.swift
//  THRAppSwift-Polarbear
//
//  Created by Brooma Service on 17/05/2019.
//  Copyright © 2019 Brooma Service. All rights reserved.
//

import Foundation
import Threads

enum CellType : String, CaseIterable {
    case Version
    case Design
    case DebugMode
    case FragmentChat
    case FullChat
    case OtherChats
    case OutsideText
    case OutsideImage
}

class CellTypeHelper {
    
    class func getCellText(forType type: CellType) -> String {
        
        switch type {
            
        case .Version:
            return String(format: "ThreadsLib - v%@\nThreadsApp - v%@", Threads.version(), PlistUtils.getAppVersion())
            
        case .Design:
            return "Design"
            
        case .DebugMode:
            return "Debug-mode"
            
        case .FragmentChat:
            return String(format: "%@ %@",
                          NSLocalizedString("show_chat_title", comment: ""),
                          NSLocalizedString("chat_fragment_title", comment: ""))
        case .FullChat:
            return String(format: "%@ %@",
                          NSLocalizedString("show_chat_title", comment: ""),
                          NSLocalizedString("chat_full_controller_title", comment: ""))
        case .OtherChats:
            return "Other Chat Integrations"
        case .OutsideText:
            return "Outside Text Message"
        case .OutsideImage:
            return "Outside Image Message"
        }
    }
    
}
